ui/src/app/views/settings/component.spec.ts

```diff
@@ -0,0 +1,77 @@
+import {ComponentFixture, fakeAsync, TestBed, tick} from '@angular/core/testing';
+import {By} from '@angular/platform-browser';
+import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
+import {RouterTestingModule} from '@angular/router/testing';
+import {ClarityModule} from '@clr/angular';
+
+import {SettingsComponent} from 'app/views/settings/component';
+import {ClusterServiceStub} from 'testing/stubs/cluster-service-stub';
+
+import {ClusterService} from 'generated';
+
+
+describe('SettingsComponent', () => {
+  let fixture: ComponentFixture<SettingsComponent>;
+  let clusterService: ClusterServiceStub;
+
+  beforeEach(fakeAsync(() => {
+    clusterService = new ClusterServiceStub();
+    TestBed.configureTestingModule({
+      declarations: [
+        SettingsComponent
+      ],
+      imports: [
+        BrowserAnimationsModule,
+        RouterTestingModule,
+        ClarityModule.forRoot()
+      ],
+      providers: [
+        { provide: ClusterService, useValue: clusterService },
+      ]}).compileComponents();
+  }));
+
+  beforeEach(fakeAsync(() => {
+    fixture = TestBed.createComponent(SettingsComponent);
+    // Trigger ngOnInit, tick() the promise, then update again to undisable the
+    // "reset" button.
+    fixture.detectChanges();
+    tick();
```

Comment:
Can you use the `updateAndTick` utility from the test helpers?

Reply:
Not sure how I feel about `updateAndTick`. The decision to do `tick()` or `detectChanges()` are going to be specific to each case, e.g. there is no particular generalization I see which implies it's going to make sense to always detectChanges, then tick() after (e.g. in the first test, you can see this isn't the case).

That said, it's of course common to do these things together. And in most cases my test shouldn't care about 0 delay timeouts.

Switched for consistency.